torch/onnx/_internal/diagnostics/infra/context.py

```diff
@@ -92,24 +94,24 @@ def sarif(self) -> sarif.Result:
         )
         return sarif_result
 
-    def with_location(self: _Diagnostic, location: infra.Location) -> _Diagnostic:
```

Comment:
Actually, you should remove the TypeVar `Diagnostic` entirely now since mypy properly supports typing_extensions.Self now.

Reply:
See the relevant comment above it's definition.

Reply:
Maybe there was a reason it was left in here? Or did you forget an instance of it because imports are failing now...